pyproject.toml

```diff
@@ -123,7 +123,7 @@ rdflib = "<7.0,>=6.0.0"
 redis = { version = ">=3.5.3,<4.2.0", optional = true }
 renku-sphinx-theme = { version = ">=0.2.0", optional = true }
 requests = ">=2.23.0,<2.28.2"
-responses = { version = ">=0.7.0,<0.22.0", optional = true }
+responses = "^0.22.0"
```

Comment:
You should always pin dependencies with some maximum version, so upstream updating doesn't suddenly break things on our end

Reply:
The reason for that is because mypy was broken in the our CI. And updating it and the responses library fixed it. But now we get more errors from mypy. The funny thing is that I do not even see those errors when I run it locally. And mypy always worked for me locally.

Not sure if you have seen this before... I will try to fix it tomorrow.

Reply:
So in mypy 0.991 or 0.990 they added a breaking change for implicit optional types. I.e. before if you just assign None as a default for an argument mypy considered that to be `Optional[x]` but not anymore. So I added the flag to get us the old behaviour. Then we can fix this up in a future PR. See the link below on the optional implicit stuff. 

https://mypy-lang.blogspot.com/2022/11/mypy-0990-released.html

Reply:
Can you create a follow-up issue to address this?

Reply:
Yes: https://github.com/SwissDataScienceCenter/renku-python/issues/3219